chain/xrp/client/client.go

```diff
@@ -0,0 +1,992 @@
+	for _, node := range affectedNodes {
+
+		if node.ModifiedNode != nil {
+			modifiedNode := node.ModifiedNode
+			if modifiedNode.LedgerEntryType == "AccountRoot" {
+
+				if node.ModifiedNode.FinalFields == nil {
+					continue
+				}
+
+				var (
+					contract     string
+					finalBalance float64
+					parseErr     error
+				)
+
+				if node.ModifiedNode.FinalFields.Balance.XRPAmount != "" {
+					contract = ""
+				} else {
+					finalBalance, parseErr = strconv.ParseFloat(node.ModifiedNode.FinalFields.Balance.TokenAmount.Value, 10)
+					if parseErr != nil {
+						return xclient.TxInfo{}, parseErr
+					}
+
+					if finalBalance > 0 {
+						contract = node.ModifiedNode.FinalFields.HighLimit.Issuer
+					} else {
+						contract = node.ModifiedNode.FinalFields.LowLimit.Issuer
+					}
+				}
+
+				convertedTransactedAmount, conversionErr := extractModifiedNodeBalance(node.ModifiedNode)
+				if conversionErr != nil {
+					return xclient.TxInfo{}, conversionErr
+				}
+
+				if modifiedNode.FinalFields.Account != txResponse.Result.Account {
+					tf.AddDestination(
+						xc.Address(modifiedNode.FinalFields.Account),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount,
+						nil,
+					)
+				} else {
+					tf.AddSource(
+						xc.Address(modifiedNode.FinalFields.Account),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount,
+						nil,
+					)
+				}
+
+			} else if node.ModifiedNode.LedgerEntryType == "RippleState" {
+
+				if node.ModifiedNode.FinalFields == nil {
+					continue
+				}
+
+				var (
+					contract        string
+					finalBalance    float64
+					transactionType string
+					parseErr        error
+				)
+
+				if node.ModifiedNode.FinalFields.Balance.XRPAmount != "" {
+					contract = ""
+				} else {
+					finalBalance, parseErr = strconv.ParseFloat(node.ModifiedNode.FinalFields.Balance.TokenAmount.Value, 10)
+					if parseErr != nil {
+						return xclient.TxInfo{}, parseErr
+					}
+
+					if finalBalance > 0 {
+						transactionType = "source"
+						contract = node.ModifiedNode.FinalFields.HighLimit.Issuer
+					} else {
+						transactionType = "destination"
+						contract = node.ModifiedNode.FinalFields.LowLimit.Issuer
+					}
+				}
+
+				convertedTransactedAmount, conversionErr := extractModifiedNodeBalance(node.ModifiedNode)
+				if conversionErr != nil {
+					return xclient.TxInfo{}, conversionErr
+				}
+
+				//if modifiedNode.FinalFields.Account != txResponse.Result.Account {
+				if transactionType == "destination" {
+					tf.AddDestination(
+						xc.Address(modifiedNode.FinalFields.LowLimit.Issuer),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount,
+						nil,
+					)
+				} else if transactionType == "source" {
+					tf.AddSource(
+						xc.Address(modifiedNode.FinalFields.HighLimit.Issuer),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount,
+						nil,
+					)
+				}
```

Comment:
I think there is a bug here where if it's an XRP amount, then `transactionType == ""` and no transfer gets updated.

I think you could simplify this logic and also not need this intermediate `transactionType` enum.  If you still need it, you should use a boolean instead.